task_manager/task_manager.hh

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+#include "replica/database.hh"
+
+using task_id = utils::tagged_uuid<struct task_id_tag>;
+using is_abortable = bool_class <struct abortable_tag>;
+
+class task_manager {
+public:
+    enum class module_type {
+        repair = 0,
+        compaction
+    };
+    static module_type get_module(std::string module) {
+        if (module == "repair") {
+            return module_type::repair;
+        } else {
+            return module_type::compaction;
+        }
+    }
+    enum class task_state {
+        running,
+        done,
+        failed
+    };
+    class task;
+    using task_ptr = seastar::shared_ptr<task_manager::task>;
+    using task_set = std::unordered_set<task_ptr>;
+    using foreign_task_ptr = seastar::foreign_ptr<task_ptr>;
+    using foreign_task_vector = std::vector<foreign_task_ptr>;
+    using task_in_module = std::pair<module_type, task_ptr>;
+    using tasks_per_module = std::map<module_type, task_set>;
+    class task {
+    public:
+        struct status {
+            task_id id;
+            std::string type;
+            task_state state;
+            gc_clock::time_point start_time;
+            gc_clock::time_point end_time;
+            std::string error;
+            int number;                     // A running sequence number of the task.
```

Comment:
Better to have an initializer here for anything doesn't have a good default constructor.